utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java

```diff
@@ -654,6 +660,111 @@ String dsmExtract(
         return "ok";
     }
 
+    @RequestMapping("/dsm/manual/produce")
+    String dsmManualCheckpointProduce(
+        @RequestParam(required = true, name = "type") String type,
+        @RequestParam(required = true, name = "target") String target,
+        HttpServletResponse response
+    ) {
+        DSMContextCarrier headers = new DSMContextCarrier();
+
+        DataStreamsCheckpointer dsmCheckpointer = DataStreamsCheckpointer.get();
+
+        dsmCheckpointer.setProduceCheckpoint(type, target, headers);
+
+        System.out.println("[DSM Manual Produce] After completion: " + headers.getData());
+
+        // Add headers that include DSM pathway context to response headers
+        for (Map.Entry<String, Object> entry : headers.entries()) {
+            response.addHeader(entry.getKey(), entry.getValue().toString());
+        }
+        return "ok";
+    }
+
+    @RequestMapping("/dsm/manual/produce_with_thread")
+    String dsmManualCheckpointProduceWithThread(
```

Comment:
what's the aim of this test/endpoint? The context propagation through an executor service is already broadly tested in the java tracer

Reply:
to ensure the api for data streams manual checkpointing is consistent across langs (in this case, that is handles multi-process code)